File: 프로그래머스/lv1/이전/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int size = nums.length/2;
        
        Set<Integer> pokemon = new TreeSet<Integer>();
        for(int n : nums){
            pokemon.add(n);
        }
        
        if(pokemon.size() >= size){
            answer = size;
        }else{
            answer = pokemon.size();
        }
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
# N 마리 중 N/2 마리 소유 가능
# 같은 종류의 폰켓몬은 동일 번호


def solution(nums):
    del_dup = set(nums)
    return len(del_dup) if len(del_dup) <= len(nums)/2 else len(nums)/2